cmd/thanos/sidecar.go

```diff
@@ -125,6 +131,40 @@ func runSidecar(
 
 		ctx, cancel := context.WithCancel(context.Background())
 		g.Add(func() error {
+			if validateProm {
+				// Retry infinitely until we get Prometheus version.
+				err := runutil.Retry(2*time.Second, ctx.Done(), func() error {
+					err := m.FetchPromVersion(logger)
+					if err != nil {
+						level.Warn(logger).Log(
+							"msg", "failed to get Prometheus version. Is Prometheus running? Retrying",
+							"err", err,
+						)
+						return errors.Wrapf(err, "fetch Prometheus version")
+					}
+					return nil
+				})
+				if err != nil {
+					return err
+				}
+
+				if m.version == nil {
```

Comment:
when that can happen? If Prometheus is wrongly build right? Happy to "skip", but can we have more info?

Reply:
Emm.. What kind of info? I think it is enough here. There are 2 situations for m.version equals nil: 
1. bad version such as **bar**、**foo** 
2. no version when building

I can add logs info for these 2 cases in func parseVersion